lib/data/repository/default_repository.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:dio/dio.dart';
+import 'package:marvel/data/data_sources/local_character_data_source.dart';
+import 'package:marvel/data/data_sources/remote_data_source.dart';
+import 'package:marvel/data/models/local_models/local_character.dart';
+import 'package:marvel/domain/entities/character.dart';
+import 'package:marvel/domain/entities/series.dart';
+import 'package:marvel/domain/exceptions/exceptions.dart';
+import 'package:marvel/domain/repository/marvel_repository.dart';
+import 'package:marvel/data/converters/converter.dart';
+
+class DefaultRepository implements MarvelRepository {
+  final DioDataSource? dioDataSource;
+  final HiveDataSource? hiveDataSource;
+  bool onceLoaded = false;
+  late int offset;
+
+  DefaultRepository({this.hiveDataSource, this.dioDataSource});
+
+  @override
+  Future<List<Character>> getCharacters(int offset) async {
+    try {
+      if (!onceLoaded) {
+        return _getCharactersFromLocalDB();
+      } else {
```

Comment:
It would be good to extract else branch into separate method similar as you did for `_getCharactersFromLocalDB()`